src/com/microsoft/azure/documentdb/JsonSerializable.java

```diff
@@ -440,9 +440,8 @@ void onSave() {
             return c.cast(this.propertyBag);
         } else {
             // POJO
-            if (!c.isMemberClass() || !Modifier.isStatic(c.getModifiers())) {
-                throw new IllegalArgumentException(
-                        "c must be a member (not an anonymous or local) and static class.");
+            if (c.isMemberClass() && !Modifier.isStatic(c.getModifiers())) {
+                throw new IllegalArgumentException("Nested classes must be static.");
```

Comment:
Thanks, but do me a favor to change to the following.

                if (c.isAnonymousClass() || c.isLocalClass()) {
                   throw new IllegalArgumentException(
                            String.format("%s can't be an anonymous or local class.", c.getName()));
                }

                if (c.isMemberClass() && !Modifier.isStatic(c.getModifiers())) {
                    throw new IllegalArgumentException(
                            String.format("%s must be static if it's a member class.", c.getName()));
                }